python/cudf/cudf/core/column/column.py

```diff
@@ -1412,11 +1341,10 @@ def as_column(arbitrary, nan_as_null=True, dtype=None, name=None):
                     data = as_column(sr, nan_as_null=nan_as_null)
                 else:
                     data = as_column(
-                        np.array(arbitrary, dtype=np_type),
+                        np.array(arbitrary, dtype=np.dtype(dtype)),
```

Comment:
```suggestion
                        np.asarray(arbitrary, dtype=np.dtype(dtype)),
```

Reply:
Go through `as_column` function and replace all instances of `np.array` with `np.asarray` where appropriate.

Reply:
Done